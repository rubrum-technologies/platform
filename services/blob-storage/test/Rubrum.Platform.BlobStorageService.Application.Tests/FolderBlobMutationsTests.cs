using CookieCrumble;
using HotChocolate.Types.Relay;
using Rubrum.Graphql;
using Rubrum.Platform.BlobStorageService.Folders;
using Shouldly;
using Xunit;
using static Rubrum.Platform.BlobStorageService.BlobStorageServiceTestConstants;

namespace Rubrum.Platform.BlobStorageService;

public sealed class FolderBlobMutationsTests : RubrumGraphqlTestBase<BlobStorageServiceApplicationTestModule>
{
    private readonly INodeIdSerializer _idSerializer;

    public FolderBlobMutationsTests()
    {
        _idSerializer = GetRequiredService<INodeIdSerializer>();
    }

    [Fact]
    public async Task ChangeName_Suscess()
    {
        var folderId = _idSerializer.Format(nameof(FolderBlob), FolderBlobId);

        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
            $$"""
               mutation {
                  changeFolderBlobName(input: {id: "{{folderId}}", name: "Test2"}) {
                      folderBlob {
                          id
                          name
                      }
                      errors {
                          __typename
                      }
                  }
               }
              """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }
}
