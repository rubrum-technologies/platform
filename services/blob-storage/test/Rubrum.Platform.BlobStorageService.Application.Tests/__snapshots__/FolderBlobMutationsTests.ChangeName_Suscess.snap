{
  "data": {
    "changeFolderBlobName": {
      "folderBlob": {
        "id": "Rm9sZGVyQmxvYjp57bnf1uXmRq/eujZQB0Tf",
        "name": "Test2"
      },
      "errors": null
    }
  }
}
